### PR TITLE
fix(protocol-designer): fix pipette type button display condition in EditInstrumentsModal

### PR DESCRIPTION
**Pull request by koji**
# Overview
This PR fixes multiple bugs in `EditInstrumentsModal`

1. pipette type button display issue.
2. Add a pipette button issue
3. swap button display condition
4. buttons for the modal

### after

https://github.com/user-attachments/assets/b7b6afbd-9ce3-44e0-b6b3-25cc5a7bf01b

close RQA-3285 RQA-3294 RQA-3296

## Test Plan and Hands on Testing
Need to check the following cases

RQA-3285 
1. create or import a protocol that has only one pipette

2. click instruments edit button

4. click edit button in the existing pipette
check there is 96-channel button then hit cancel

5. click add a pipette button
check there is no 96-channel button (because of the existing pipette)

6. add a pipette single or multi

7. click edit for each mount
check there is no 96-channel button 

8. remove both mounts and click a pipette button 
there is 96-channel button (2 add a pipette buttons will show up which is a bug and it will be resolved by a following pr) 

Add a pipette button
only one button is always placed below a pipette info card
when clicking swap, add a pipette button is placed the same position

## Changelog
- add a new util function and its tests
- update text in Edit Instruments modal and test

## Risk assessment
low

### CODE REVIEW

File: protocol-designer/src/organisms/EditInstrumentsModal/index.tsx [hunk 1]
```diff
@@ -252,67 +262,54 @@ export function EditInstrumentsModal(
...
+                  pipetteName={leftPipette.name}
+                  tiprackDefURIs={leftPipette.tiprackDefURI}
                   editClick={() => {
                     setPage('add')
                     setMount('left')
                     setPipetteType(leftInfo.type)
                     setPipetteGen(leftInfo.gen)
                     setPipetteVolume(leftInfo.volume)
-                    setSelectedTips(leftPip.tiprackDefURI)
+                    setSelectedTips(leftPipette.tiprackDefURI as string[])
```
Comment (ncdiehl11) [inline protocol-designer/src/organisms/EditInstrumentsModal/index.tsx:266]: Is TypeScript mad if we don't cast? Needed for null checking?

File: protocol-designer/src/organisms/EditInstrumentsModal/index.tsx [hunk 2]
```diff
@@ -252,67 +262,54 @@ export function EditInstrumentsModal(
...
                   }}
                   cleanForm={() => {
-                    dispatch(deletePipettes([leftPip.id]))
+                    dispatch(deletePipettes([leftPipette.id as string]))
```
Comment (ncdiehl11) [inline protocol-designer/src/organisms/EditInstrumentsModal/index.tsx:269]: same

File: protocol-designer/src/organisms/EditInstrumentsModal/index.tsx [hunk 3]
```diff
@@ -252,67 +262,54 @@ export function EditInstrumentsModal(
...
+                  pipetteName={rightPipette.name}
+                  tiprackDefURIs={rightPipette.tiprackDefURI}
                   editClick={() => {
                     setPage('add')
                     setMount('right')
                     setPipetteType(rightInfo.type)
                     setPipetteGen(rightInfo.gen)
                     setPipetteVolume(rightInfo.volume)
-                    setSelectedTips(rightPip.tiprackDefURI)
+                    setSelectedTips(rightPipette.tiprackDefURI as string[])
```
Comment (ncdiehl11) [inline protocol-designer/src/organisms/EditInstrumentsModal/index.tsx:287]: same for all string casting

File: protocol-designer/src/organisms/EditInstrumentsModal/utils.ts [hunk 4]
```diff
@@ -28,3 +33,30 @@ export const getSectionsFromPipetteName = (
     volume,
   }
 }
+
+export const shouldShowPipetteType = (
```
Comment (ncdiehl11) [inline protocol-designer/src/organisms/EditInstrumentsModal/utils.ts]: nit, maybe `getShouldShowPipetteType`

File: protocol-designer/src/organisms/EditInstrumentsModal/utils.ts [hunk 5]
```diff
@@ -28,3 +33,30 @@ export const getSectionsFromPipetteName = (
...
+  type: PipetteType,
+  has96Channel: boolean,
+  leftPipette?: PipetteOnDeck | null,
+  rightPipette?: PipetteOnDeck | null,
+  currentEditingMount?: PipetteMount | null
+): boolean => {
+  if (type === '96') {
+    // if a protocol has 96-Channel, no 96-Channel button
+    if (has96Channel) return false
```
Comment (ncdiehl11) [inline protocol-designer/src/organisms/EditInstrumentsModal/utils.ts]:
```suggestion
    if (has96Channel) {
        return false
    }
```

File: protocol-designer/src/organisms/PipetteInfoItem/index.tsx [hunk 6]
```diff
@@ -40,9 +41,9 @@ export function PipetteInfoItem(props: PipetteInfoItemProps): JSX.Element {
           <StyledText desktopStyle="bodyDefaultSemiBold">
             {i18n.format(
               t('pip', {
-                mount: is96Channel ? t('left_right') : t(`${mount}`),
+                mount: is96Channel ? t('left_right') : mount,
               }),
-              'capitalize'
+              'titleCase'
```
Comment (ncdiehl11) [inline protocol-designer/src/organisms/PipetteInfoItem/index.tsx:46]: 👍

**Review by ncdiehl11 — APPROVED** [review]
looks good! couple of comments/questions

File: protocol-designer/src/organisms/EditInstrumentsModal/index.tsx (hunk 1 above)
Comment (koji) [reply protocol-designer/src/organisms/EditInstrumentsModal/index.tsx:266]: VSCode shows an error. Maybe only on my end 🤔

**Comment by koji** [comment]
- [x] address feedback
- [x] resolve conflicts

**Pull request merged**